src/App.tsx

```diff
@@ -1,14 +1,43 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import './App.scss';
 
+import { useDispatch } from 'react-redux';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+// import { useAppDispatch } from './app/hooks';
+import { getTodos } from './api';
+import { actions as todoActions } from './features/todos';
+import { useAppSelector } from './app/hooks';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+  const [hasReceivedTodos, setHasReceivedTodos] = useState(false);
```

Comment:
It's a redundant state. You could determine if you need to render the `<TodoList />` component using just a constant: `const toShowTodoList = !isLoading && !hasError;`